feature-libs/customer-ticketing/components/customer-ticketing-list/customer-ticketing-list.component.ts

```diff
@@ -1,11 +1,163 @@
+    },
+    sorts: [
+      { code: 'byId', selected: true },
+      { code: 'byChangedDate', selected: false },
+    ],
+    tickets: [
+      {
+        availableStatusTransitions: [
+          {
+            id: 'CLOSED',
+            name: 'Closed',
+          },
+        ],
+        id: '0000001',
+        createdAt: '2021-01-13T10:06:57+0000',
+        modifiedAt: '2021-01-13T10:06:57+0000',
+        status: {
+          id: 'CLOSED',
+          name: 'Closed',
+        },
+        subject: 'My drill is broken.',
+        ticketCategory: {
+          id: 'ENQUIRY',
+          name: 'Enquiry',
+        },
+        ticketEvents: [
+          {
+            author: 'Mark Rivers',
+            createdAt: '2021-01-13T10:06:57+0000',
+            message:
+              'It is broken when I receive it. Please send one replacement to me.',
+            toStatus: {
+              id: 'CLOSED',
+              name: 'Closed',
+            },
+          },
+        ],
+      },
+      {
+        availableStatusTransitions: [
+          {
+            id: 'CLOSED',
+            name: 'Closed',
+          },
+        ],
+        id: '0000002',
+        createdAt: '2021-01-14T10:06:57+0000',
+        modifiedAt: '2021-01-14T10:06:57+0000',
+        status: {
+          id: 'OPEN',
+          name: 'Open',
+        },
+        subject: 'Need fix for my door',
+        ticketCategory: {
+          id: 'ENQUIRY',
+          name: 'Enquiry',
+        },
+        ticketEvents: [
+          {
+            author: 'Bob',
+            createdAt: '2021-01-14T10:06:57+0000',
+            message: 'Door received broken',
+            toStatus: {
+              id: 'OPEN',
+              name: 'Open',
+            },
+          },
+        ],
+      },
+    ],
+  });
+
+  tickets = this.tickets$.pipe(
+    map((ticketsList) => {
+      return ticketsList.tickets.map((ticket) => {
+        return {
+          id: ticket.id,
+          subject: ticket.subject,
+          ticketCategory: ticket.ticketCategory.name,
+          status: ticket.status,
+          createdAt: ticket.createdAt,
+          modifiedAt: ticket.modifiedAt,
+        };
+      });
+    })
+  );
+
+  ngOnInit(): void {
+    this.tickets$.subscribe((tickets) => {
+      this.customerTicketsFlag = tickets && tickets.tickets?.length > 0;
+    });
+  }
+
+  getSortLabels(): Observable<{ byId: string; byChangedDate: string }> {
+    return combineLatest([
+      this.translation.translate('customerTicketing.ticketId'),
+      this.translation.translate('customerTicketing.changedOn'),
+    ]).pipe(
+      map(([textById, textByChangedDate]) => {
+        return {
+          byId: textById,
+          byChangedDate: textByChangedDate,
+        };
+      })
+    );
+  }
+
+  changeSortCode(sortCode: string): void {
+    this.sortType = sortCode;
+    this.pageChange(0);
+  }
+
+  pageChange(page: number): void {
+    this.createTicketListEvent(this.sortType, page);
+    // this.fetchTicketList(event);
```

Comment:
Please remove

Reply:
This will be used in the next PR with backend calls